### DISCUSSION TITLE
[Fooocus 2.2.0-rc1] Metadata, Isolation, Array Support, Docker

### DISCUSSION THREAD

**Discussion by mashb1t** [lllyasviel/Fooocus#2362] [General]
## What's Changed
* fix: sort with casefold, case insensitive by @mashb1t
* feat: add early return for prompt expansion when no new tokens should be added by @mashb1t
* feat: ignore DS_Store by @charliewilco in https://github.com/lllyasviel/Fooocus/pull/2313
* feat: advanced params refactoring + prevent users from skipping/stopping other users tasks in queue by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/981
* feat: add list of 100 most popular animals to wildcards by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/985
* feat: add advanced parameter for disable_intermediate_results (progress_gallery) by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/1013
* feat: add ability to load checkpoints and loras from multiple locations by @dooglewoogle in https://github.com/lllyasviel/Fooocus/pull/1256
* feat: allow users to specify the number of threads when running on CPU by @maxim-saplin in https://github.com/lllyasviel/Fooocus/pull/1601
* feat: improve bug report and feature request issue templates by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/1631
* fix: correctly create directory for path_outputs if not existing by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/1668
* fix: allow path_outputs to be outside of root dir by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/2332
* feat: add button to enable LoRAs by @MindOfMatter in https://github.com/lllyasviel/Fooocus/pull/2210
* feat: make lora number editable in config by @MindOfMatter in https://github.com/lllyasviel/Fooocus/pull/2215
* feat: make lora min max weight editable in config by @MindOfMatter in https://github.com/lllyasviel/Fooocus/pull/2216
* feat: add array support on main prompt by @flannerybh in https://github.com/lllyasviel/Fooocus/pull/1503
* feat: use consistent file name in gradio by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/1932
* feat: add metadata to images by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/1940
* feat: add jpg and webp support, add exif data handling for metadata by @mashb1t and @eddyizm in https://github.com/lllyasviel/Fooocus/pull/1863
* feat: add docker files by @whitehara and @mashb1t in https://github.com/lllyasviel/Fooocus/pull/1418

## New Contributors
* @charliewilco made their first contribution in https://github.com/lllyasviel/Fooocus/pull/2313
* @dooglewoogle made their first contribution in https://github.com/lllyasviel/Fooocus/pull/1256
* @maxim-saplin made their first contribution in https://github.com/lllyasviel/Fooocus/pull/1601
* @MindOfMatter made their first contribution in https://github.com/lllyasviel/Fooocus/pull/2210
* @flannerybh made their first contribution in https://github.com/lllyasviel/Fooocus/pull/1503
* @whitehara made their first contribution in https://github.com/lllyasviel/Fooocus/pull/1418

**Full Changelog**: https://github.com/lllyasviel/Fooocus/compare/2.1.865...2.2.0-rc1

<hr /><em>This discussion was created from the release <a href='https://github.com/lllyasviel/Fooocus/releases/tag/2.2.0-rc1'>2.2.0-rc1</a>.</em>

**Comment by mashb1t** [comment]
It would be highly appreciated if you could help test the develop branch for potential bugs. Thank you for your support!

DISCLAIMER: this release and the milestone have not been initiated / confirmed by @lllyasviel and include unapproved but highly requested features, which have been tested and are fully functional.

Milestone for reference: https://github.com/lllyasviel/Fooocus/milestone/1

**Reply by eddyizm** [reply]
I'll test this tonight!

**Reply by mashb1t** [reply]
I've already spotted and eliminated 2 bugs in develop, hope they were the last ones 🤞

**Comment by f0n51** [comment]
I maybe found a bug with the LoRA dropdowns:

I'm running in a Google Colab machine with the following code:
```!pip install pygit2==1.12.2
!rm -rf /content/Fooocus
%cd /content
!git clone -b develop https://github.com/lllyasviel/Fooocus.git
%cd /content/Fooocus
!python entry_with_update.py --share --always-high-vram --all-in-fp16
```

Then click Advanced - Model and open the LoRA 1 dropdown with the sd_xl_offset_example-lora_1:
![image](https://github.com/lllyasviel/Fooocus/assets/155714366/536efde8-de60-44b0-af55-5cdd833f2a18)

The dropdown itself only contains a 'None' value and after choosing 'None' and opening the dropdown again, the sd_xl_offset_example-lora_1 is gone forever:
![image](https://github.com/lllyasviel/Fooocus/assets/155714366/3b10d9bc-9573-4891-91e1-b5e3848dcd0b)

It works again after clicking 'Refresh all files'.

**Reply by mashb1t** [reply]
This is i think a general issue already present in the main branch. Please double-check it there
We might have to add an initial reload after lora/model download and first "cold boot"

**Reply by mashb1t** [reply]
=> @f0n51 this is now a separate issue, see https://github.com/lllyasviel/Fooocus/issues/2372

**Reply by f0n51** [reply]
Yeah,  I wasn't sure about creating a separate issue for this as I still have not checked the main branch concerning the LoRA dropdowns. 
But thanks for creating the issue, I try to check the main branch today :)

**Comment by barepixels** [comment]
Very happy to see metadata in png.  Thanks mashb1t and lllyasviel.  One thing I notice is that you have to remember to activate it on every time you launch Fooocus.  Wouldn't be be better to set it in config.txt?

**Reply by mashb1t** [reply]
You can already activate it in the config. Please find the keys/values in the config tutorial / config example file.

**Reply by eddyizm** [reply]
I have to test this but my plan was to add it to one of my presets.

**Comment by toutjavascript** [comment]
Hello

I don't understand how to install it on windows, in a distinct folder to make tests...

**Reply by mashb1t** [reply]
Basically you only have to switch to develop and pull (just to ensure you have the latest version of develop).

```
git checkout develop
git pull
```

There are no package changes and no risk to permanently change anything if you back up your config.txt before, so going back to main is as simple as running `git checkout main` again after testing.

If you want to test it in a separate folder then please download the [7z file](https://github.com/lllyasviel/Fooocus/releases/download/release/Fooocus_win64_2-1-831.7z), run `run.bat` and follow the same steps to checkout and pull develop as mentioned above.

**Comment by toutjavascript** [comment]
thanks @mashb1t 

It is obvious in fact ! But I am so bad with Git.

**Comment by mashb1t** [comment]
ETA tomorrow, 02.03.
